### DISCUSSION TITLE
How to run site with "http://localhost/example' instead of "http://example.test" while using Laravel Herd

### DISCUSSION THREAD

**Discussion by ayoubelmoudni** [beyondcode/herd-community#438] [Q&A] [answered]
I have a project that needs to be run as "http://localhost/example" and not "http://example.test" in my local, am using Herd and I like it, is there any way to do this?

**Comment by mpociot** [comment, marked as answer]
The easiest way to achieve this is by isolating the site - you can do this by running "herd isolate 8.3" (or whatever version you need) in your terminal in the project root. 

This gives you a custom nginx config file, which you can adjust to your specific needs. The config file can be found in ~/Library/Application Support/Herd/config/valet/Nginx

**Reply by ayoubelmoudni** [reply]
Thanks

**Reply by LouiScura** [reply]
Do I need to restart anything after updating the file? I change the 'listen' parameter to in my case localhost:8000 but I only get the Herd 404 page "Site not found Need some help? Check out the [docs](https://herd.laravel.com/docs).". 

`# ISOLATED_PHP_VERSION=8.3
server {
    listen localhost:8000;`

**Reply by antwanvdm** [reply]
I followed the isolate step as well, but I don't even get a specific nginx file. I don't want a different host, but just want to add config to nginx which I previously had in a .htaccess when using apache. Any advice in direction I should take to edit a nginx config for one specific website?

**Comment by 9025351798** [comment]
<?php

session_start(); // Start the session to track check-in/out status

$file = "checkins.txt";

function recordCheckIn($name) {
    global $file;
    $timestamp = date("Y-m-d H:i:s");
    $data = "$name,in,$timestamp\n";
    file_put_contents($file, $data, FILE_APPEND);
    $_SESSION['checked_in'] = true; // Set session variable
    $_SESSION['name'] = $name;  // Store name in session
    return "Check-in recorded for $name at $timestamp.";
}

function recordCheckOut($name) {
    global $file;
    $timestamp = date("Y-m-d H:i:s");
    $data = "$name,out,$timestamp\n";
    file_put_contents($file, $data, FILE_APPEND);
    unset($_SESSION['checked_in']); // Clear session variable
    unset($_SESSION['name']); // Clear name from session
    return "Check-out recorded for $name at $timestamp.";
}

if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $name = $_POST["name"];

    if (isset($_POST['check_in'])) {
        $message = recordCheckIn($name);
    } elseif (isset($_POST['check_out']) && isset($_SESSION['checked_in']) && $_SESSION['name'] == $name ) { //check if they are checked in before checkout
        $message = recordCheckOut($name);
    }
    else if(isset($_POST['check_out']) && (!isset($_SESSION['checked_in']) || $_SESSION['name'] != $name)){
        $message = "You can't check out before checking in, or you have entered a different name.";
    }


}

?>

<!DOCTYPE html>
<html>
<head>
    <title>Check-in/Check-out</title>
</head>
<body>

<h1>Check-in/Check-out</h1>

<form method="post">
    Name: <input type="text" name="name" required><br>

    <?php if (!isset($_SESSION['checked_in'])): ?>
        <input type="submit" name="check_in" value="Check In">
    <?php else: ?>
        <input type="submit" name="check_out" value="Check Out">
    <?php endif; ?>
</form>

<?php if (isset($message)): ?>
    <p><?php echo $message; ?></p>
<?php endif; ?>

</body>
</html>

**Comment by 9025351798** [comment]
<?php

session_start(); // Start the session to track check-in/out status

$file = "checkins.txt";

function recordCheckIn($name) {
    global $file;
    $timestamp = date("Y-m-d H:i:s");
    $data = "$name,in,$timestamp\n";
    file_put_contents($file, $data, FILE_APPEND);
    $_SESSION['checked_in'] = true; // Set session variable
    $_SESSION['name'] = $name;  // Store name in session
    return "Check-in recorded for $name at $timestamp.";
}

function recordCheckOut($name) {
    global $file;
    $timestamp = date("Y-m-d H:i:s");
    $data = "$name,out,$timestamp\n";
    file_put_contents($file, $data, FILE_APPEND);
    unset($_SESSION['checked_in']); // Clear session variable
    unset($_SESSION['name']); // Clear name from session
    return "Check-out recorded for $name at $timestamp.";
}

if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $name = $_POST["name"];

    if (isset($_POST['check_in'])) {
        $message = recordCheckIn($name);
    } elseif (isset($_POST['check_out']) && isset($_SESSION['checked_in']) && $_SESSION['name'] == $name ) { //check if they are checked in before checkout
        $message = recordCheckOut($name);
    }
    else if(isset($_POST['check_out']) && (!isset($_SESSION['checked_in']) || $_SESSION['name'] != $name)){
        $message = "You can't check out before checking in, or you have entered a different name.";
    }


}

?>

<!DOCTYPE html>
<html>
<head>
    <title>Check-in/Check-out</title>
</head>
<body>

<h1>Check-in/Check-out</h1>

<form method="post">
    Name: <input type="text" name="name" required><br>

    <?php if (!isset($_SESSION['checked_in'])): ?>
        <input type="submit" name="check_in" value="Check In">
    <?php else: ?>
        <input type="submit" name="check_out" value="Check Out">
    <?php endif; ?>
</form>

<?php if (isset($message)): ?>
    <p><?php echo $message; ?></p>
<?php endif; ?>

</body>
</html>
